File: databinding/src/main/java/com/example/anvil/databinding/MainActivity.java
package com.example.anvil.databinding;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

import java.lang.reflect.Constructor;

public class MainActivity extends Activity {

    private boolean home = true;

    @Override
    public void onCreate(Bundle b) {
        super.onCreate(b);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);

        setContentView(new MainLayout(this));
    }

    @Override
    public void onBackPressed() {
        if (home) {
            super.onBackPressed();
        } else {
            home = true;
            setContentView(new MainLayout(this));
        }
    }

    public void route(int pos) {
        try {
            Class cls = Class.forName(getPackageName()+"."+MainLayout.LAYOUTS[pos]);
            Constructor constructor = cls.getConstructor(Context.class);
            setContentView((View) constructor.newInstance(this));

            home = false;
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/MainActivity.kt
package com.example.anvil.databinding

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager

import java.lang.reflect.Constructor

class MainActivity : Activity() {

    var home = true

    override fun onCreate(b: Bundle?) {
        super.onCreate(b)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN)

        setContentView(MainLayout(this))
    }

    override fun onBackPressed() {
        if (home) {
            super.onBackPressed()
        } else {
            home = true
            setContentView(MainLayout(this))
        }
    }

    fun route(pos: Int) {
        try {
            val cls = Class.forName(getPackageName() + "." + LAYOUTS[pos])
            val constructor = cls.getConstructor(Context::class.java)
            setContentView(constructor.newInstance(this) as View)

            home = false
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
